woocommerce/woocommerce-android
# Feature/login local notification ii

Part II: #6989

Part I is already merged and can be [checked here](https://github.com/woocommerce/woocommerce-android/pull/6990#event-7038429826). 

### Description
Iteration 1 pe5sF9-h9-p2 for login scheduled notifications:

- Display notification after 24h of the user experiencing an error when introducing their site address
- Notifications will have 2 CTAs
   - Contact Support
   - Login with WordPress.com
- Adds tracking for notification interactions

These changes are hidden under a feature flag until we confirm we want to activate this feature in Sprint 2022.14

### Testing instructions
- Open login flow
- Click login with site address and input a "non wordpress" site address. Like `google.com`
- Click continue
-  Wait around 5 secs for the notification to appear. 
- Check logcat `login_local_notification_displayed, Properties: {"type":"site_address_error","is_debug":true}`
- Click on "Contact Support" CTA and check it opens help/support section. NOTE: notification may appear collapsed, so buttons might not be visible until you expand. 


- Repeat the same flow and check that the notification is not shown again.
- Clear app data and repeat the same flow to display the notification again
- Click on "Login with WordPress.com" and check it opens Login with WordPress.com screen
- Check logcat for: `login_local_notification_tapped, Properties: {"type":"site_address_error","is_debug":true}`


- Repeat flow to make notification appear again by clearing app data
- Dismiss the notification
- Check logcat: `login_local_notification_dismissed, Properties: {"is_debug":true}`

### Images/gif

https://user-images.githubusercontent.com/2663464/180444011-cc85cd91-c1d6-4d36-8868-cea072094dbf.mp4
